File: BehavioralPatterns/Command/Order.py
from abc import ABC, abstractmethod
from KitchenWorker import KitchenWorker

class Order(ABC):
    def __init__(self, receiver: KitchenWorker):
        self.name = None
        self.receiver = receiver
    
    def sendOrder(self):
        self.receiver.finishOrder()

class DrinkOrder(Order):
    def __init__(self, receiver: KitchenWorker):
        super().__init__(receiver)
        self.name = "drinkOrder"

class SnakeOrder(Order):
    def __init__(self, receiver: KitchenWorker):
        super().__init__(receiver)
        self.name = "snakeOrder"

File: DESIGN-CSharp/Command/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Command
{
    public abstract class Order
    {
        //廚房工作者(receiver)
        protected KitchenWorker receiver;
        
        public string name { get; set; }

        public Order(KitchenWorker receiver)
        {
            this.receiver = receiver;
        }

        //將訂單送給廚房人員
        public void sendOrder() 
        {
            this.receiver.finishOrder();
        }
    }

    public class DrinkOrder : Order
    { 
        public DrinkOrder(KitchenWorker receiver) : base(receiver) 
        {
            base.name = "drinkOrder";
        }
    }

    public class SnakeOrder : Order
    {
        public SnakeOrder(KitchenWorker receiver) : base(receiver)
        {
            base.name = "snakeOrder";
        }
    }

}
